api/v1alpha1/circuitbreaker_types.go

```diff
@@ -7,14 +7,24 @@ package v1alpha1
 
 // CircuitBreaker defines the Circuit Breaker configuration.
 type CircuitBreaker struct {
+	// Settings related to both TCP and HTTP upstream connections
```

Comment:
```suggestion
	// Settings related to both TCP upstream connections
```

Reply:
TCP settings like `MaxConnections` will apply to both TCP (e.g. TLS Passthrough) and HTTP connections. The HTTP struct contains settings that are relevant strictly to HTTP connections.

Reply:
in that case, the top level `tcp` and `http` doesnt seem very useful here and can be flattened into one, wdyt ?

Reply:
no problem. I'll update the API PR accordingly.